CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+# Contributing
+
+:+1::tada: First off, thanks for taking the time to contribute! :tada::+1:
+
+We welcome contributions via both the raising of issues and submitting pull requests. But before you
+do, please take a moment to consult the below.
+
+## Chatting
+
+If before any of the below you'd like to discuss an issue or some code, then come and chat with
+us on one of the following platforms:
+
+* [Users mailing list](https://groups.google.com/a/apereo.org/forum/#!forum/equella-users) - Most
+  active, best for more usage centric questions
+* [Developer mailing list](https://groups.google.com/a/apereo.org/forum/#!forum/equella-dev) - For
+  discussing work on the code
+* \#Equella channel in [Apereo's slack](https://apereo.slack.com/)
+
+## Submitting Issues
+
+Issues relating to the documentation are welcome. (For any issues relating to openEQUELLA itself,
+please switch over to its repo.)
+
+When raising issues, please try and be as detailed as possible. Links, line numbers, etc always
+welcome.
+
+## Contributing Code / Pull Requests
+
+As per standard Apereo projects:  If you wish to contribute to Open EQUELLA itself, you should first
+subscribe to the Open EQUELLA Google Group as outlined above and discuss your contributions with
+the other developers. You will also need to sign a
+[Contributor License Agreement](https://www.apereo.org/node/676).
+
+But after that, as per usual GitHub, please first fork the project and then branch off `master` or a
+stable branch (see below) to use a 'feature' branch.
+
+Please ensure you provide [quality commit messages](https://chris.beams.io/posts/git-commit/),
+and in the PR please detail what testing you have undertaken.
+
+## Setting up a development environment
+
+The site is served up by GitHub pages and is done so via a static [Jekyll](https://jekyllrb.com/)
```

Comment:
Using Docker can simplify setup a one liner

```sh
docker run -t --rm -v "$PWD":/usr/src/app -p "4000:4000" starefossen/github-pages
```

https://hub.docker.com/r/starefossen/github-pages

Reply:
Indeed. But not everyone (especially those doing doco) are going to have (nor want) docker. So I'll leave that for now.

Reply:
Yes, but the same argument could hold for Ruby (possibly to a lesser extent). :)

I would suggest 3 methods for contribution:
small changes - use a text editor and preview in the PR on github
Local preview with Docker - Add Christian's one liner here
Local preview with Ruby - continue with the documentation below